tests/test_curves.py

```diff
@@ -843,3 +843,57 @@ def test_compute_ndarray_error(self):
         c = SplineXYZCurve(X=R * np.cos(phi), Y=R * np.sin(phi), Z=np.zeros_like(phi))
         with pytest.raises(TypeError):
             c.compute("length", grid=np.linspace(0, 1, 10))
+
+    @pytest.mark.unit
+    def test_discontinuous_splines(self):
+        """Test splines that have break points."""
+
+        def test(method, data_key, include_breaks=True):
+            R = 2
+            phi = 2 * np.pi * np.linspace(0, 1, 1001, endpoint=True) ** 2
+
+            discontinuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                break_indices=break_indices,
+                method=method,
+            )
+
+            continuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                break_indices=None,
+                method=method,
+            )
+
+            assert discontinuous.method == method
+            assert continuous.method == method
+
+            discon_data = discontinuous.compute(data_key)[data_key]
+            cont_data = continuous.compute(data_key)[data_key]
+
+            if include_breaks:
+                np.testing.assert_allclose(discon_data, cont_data, rtol=1e-3)
+            else:
+                np.testing.assert_allclose(
+                    discon_data[~np.array(break_indices)],
```

Comment:
see comment above, I'm not sure the indexing is correct here if knots and the query points are different

Reply:
I'm actually confused why this works, because the default grid should have about 4x as many nodes as the curve has knots. It would be best to explicitly pass in the grid that you are computing on, and choose a grid that doesn't align with the spline knots.

Reply:
How do you recommend I get the indices of break points in `s`? I tried doing
```python
is_break_point = np.isin(s, discontinuous.knots[break_indices])
```
but there is only 1 `True` in `is_break_point` when I was expected 4. Does this mean the break points are not the same value in `s`?

Reply:
@ddudt So to be clear, you're saying to pass in a grid to the compute functions that is not `LinearGrid(zeta=knots)`?

Reply:
@kianorr correct